data-imports/import_hpd_violations.py

```diff
@@ -0,0 +1,198 @@
+import calendar
+import clean_utils
+import datetime
+import luigi
+import luigi_utils
+import os
+import tempfile
+import utils
+
+class DownloadHPDViolations():
+    base_url = 'http://www1.nyc.gov/assets/hpd/downloads/misc/'
+    base_out = '/root/heatseek/hpd_violations/'
+    zip_filename_format = 'Violations{date:%Y%m%d}'
+    txt_filename_format = 'Violation{date:%Y%m%d}'
+    bad_zip_filename_date = datetime.date(2015, 3, 1)
+    v2_start_date = datetime.date(2013, 5, 1)
+
+    @classmethod
+    def is_v2(cls, date):
+        return date > cls.v2_start_date
+
+    @classmethod
+    def getHPDViolationDownloadTask(cls, date):
+        if date != cls.bad_zip_filename_date:
+            zip_filename = cls.zip_filename_format.format(date=date) + '.zip'
+        else:
+            zip_filename = cls.txt_filename_format.format(date=date) + '.zip'
+
+        txt_date = date - datetime.timedelta(days=1)
+        txt_filename = cls.txt_filename_format.format(date=txt_date) + '.txt'
+
+        zip_url = cls.base_url + zip_filename
+        zipout = cls.base_out + zip_filename
+
+        return luigi_utils.DownloadUnzipTask(url=zip_url,
+                zipout=zipout,
+                zipfiles=[txt_filename],
+                zipfilesout=[cls.base_out + txt_filename])
+
+
+class ImportHPDViolationUpdate(luigi.Task):
+    date = luigi.MonthParameter()
+
+    table_name = 'hpd_violations'
+
+    # Commented-out lines are present in latest drops but not original
+    dtype_dict = {
+            'ViolationID': 'object',
+            'BuildingID': 'object',
+            'RegistrationID': 'object',
+            'Boro': 'object',
+            'HouseNumber': 'object',
+            'LowHouseNumber': 'object',
+            'HighHouseNumber': 'object',
+            'StreetName': 'object',
+            'StreetCode': 'object',
+            'Zip': 'object',
+            'Block': 'object',
+            'Lot': 'object',
+            'Class': 'object',
+            'InspectionDate': 'object',
+            'OriginalCertifyByDate': 'object',
+            'OriginalCorrectByDate': 'object',
+            'NewCertifyByDate': 'object',
+            'NewCorrectByDate': 'object',
+            'CertifiedDate': 'object',
+            'OrderNumber': 'object',
+            'NOVID': 'object',
+            'NOVDescription': 'object',
+            'NOVIssuedDate': 'object',
+            'CurrentStatus': 'object',
+            'CurrentStatusDate': 'object'
+            # 'BoroID': 'object',
+            # 'Apartment': 'object',
+            # 'Story': 'object',
+            # 'ApprovedDate': 'object',
+            # 'CurrentStatusID': 'int64',
```

Comment:
See the comment on line 70 - they weren't included in the first v2 (txt / csv) updates from 2013, but are in the most recent updates. I didn't figure out when they started getting included. I guess I could try something like creating a method that returns different dtype_dicts for a provided date, but then we'd need to also separate database schema specification from the csv data description. Certainly doable, and maybe worth doing, given that I've been running into memory limits when trying to do stuff with this violations data.